extensions/oplogPopulator/modules/Allocator.js

```diff
@@ -114,6 +127,7 @@ class Allocator {
             const connector = this._bucketsToConnectors.get(bucket);
             if (connector) {
                 await connector.removeBucket(bucket);
+                this.emit('bucket-removed', bucket, connector);
```

Comment:
similarly, this should be done the other way to avoid race conditions:

```suggestion
                this.emit('bucket-removed', bucket, connector);
                await connector.removeBucket(bucket);
```

Reply:
(to avoid these, maybe we should [in a later PR] refactor the code to make these calls non-async, since currently they do not perform the change...)

Reply:
Done